frontend/public/components/storage/attach-storage.tsx

```diff
@@ -0,0 +1,437 @@
+        return (
+          storage.attach.allContainers ||
+          storage.attach.containers[container.name]
+        );
+      };
+
+      //   Look at the existing mount paths so that we can warn if the new value is not unique.
+      storage.existingMountPaths = getMountPaths(
+        _.get(resourceObj.spec, "template"),
+        isContainerSelected
+      );
+      var updateVolumeName = function() {
+        var pvc = _.get(resourceObj.spec, "volumes.persistentVolumeClaim");
+        if (!pvc) {
+          return;
+        }
+
+        // Check if there is already a volume for this PVC.
+        var volumes = _.get(resourceObj.spec, "template.spec.volumes");
+        var volume = _.find(volumes, {
+          persistentVolumeClaim: {
+            claimName: pvc.metadata.name
+          }
+        });
+        if (volume) {
+          // If there's already a  volume, reuse the volume name.
+          storage.attach.volumeName = resourceObj.spec.volume.metadata.name;
+          storage.volumeAlreadyMounted = true;
+        } else if (storage.volumeAlreadyMounted) {
+          // Clear the volume name value since it was associated with the previously selected PVC.
+          storage.attach.volumeName = "";
+          storage.volumeAlreadyMounted = false;
+        }
+      };
+      storage.onPVCSelected = updateVolumeName;
+
+      // generate a volume name if not provided
+      if (!volumeName) {
+        // Create a volume name using the project
+        volumeName = "volume-" + metadata.namespace;
+      }
+      if (mountPath) {
+        // for each container in the pod spec, add the new volume mount
+
+        _.forEach(resourceObj.spec.template.spec.containers, function(
+          container
+        ) {
+          if (isContainerSelected(container)) {
+            var newVolumeMount = createVolumeMount(
+              volumeName,
+              mountPath,
+              subPath,
+              readOnly
+            );
+            if (!container.volumeMounts) {
+              container.volumeMounts = [];
+            }
+            container.volumeMounts.push(newVolumeMount);
+          }
+        });
+      }
+
+      // add the new volume to the pod template
+      if (!storage.volumeAlreadyMounted) {
+        resourceObj.spec.volumes = resourceObj.spec.volumes || [];
+        resourceObj.spec.volumes.push(
+          createVolume(volumeName, storage.attach.persistentVolumeClaim)
+        );
+      }
+
+      e.preventDefault();
+      this.setState({ inProgress: true });
+      this.setState({ resourceObj });
+      k8sUpdate(ko, resourceObj, metadata.namespace, metadata.name).then(
+        resource => {
+          this.setState({ inProgress: false });
+          history.push(resourceObjPath(resource, referenceFor(resource)));
+        },
+        err => this.setState({ error: err.message, inProgress: false })
+      );
+    }
+    render() {
+      const title = `${this.props.titleVerb} Storage`;
```

Comment:
Isn't this always Add Storage?

Reply:
Yes, for now, I was not sure if we would ever use this form for editing

Reply:
Let's hard code it to Add Storage for now, and we can parameterize later when needed